src/main/java/com/chess/backend/gamemodel/Move.java

```diff
@@ -2,6 +2,8 @@
 
 import com.chess.backend.gamemodel.constants.PieceType;
 
+import java.util.Objects;
+
 public class Move {
```

Comment:
i would suggest to make the class static, because the class has no actual state, it should only manipulate the board.
and the class will only have executeMove methods which will get all the parameters needed

Reply:
I do not really get your point. I would not make this class static, because it needs to be instantiated in other classes e.g. Piece with get all moves. But maybe I am getting you wrong here

Reply:
okay got your idea.